In [4]:
import pandas as pd
df = pd.read_csv("high_blood_pressure.csv")

목표: 고혈압 환자 치료 전후의 혈압

효과가 있는지 대응 표본 t-검정을 진행하시외.

귀무가설(H0):  
  >= 0
대립가설(H1): μ
  
μ
  = (치료 후 혈압 - 치료 전 혈압)의 평균
유의
수준0.05
μ
 의 표본평균은?(소수 둘째자리까지 반올림)
검정통계량 값은?(소수 넷째자리까지 반올림)
p-값은?(소수 넷째자리까지 반올림)
가설검정의 결과는

대응 표본 T검정: ttest_rel? (유의수준 5%)

In [5]:
df.head()

Id     sex  age  bp_pre  bp_post
0  p001    Male   33     149      129
1  p002    Male   39     168      168
2  p003    Male   70     176      155
3  p004  Female   41     169      178
4  p005    Male   48     160      126

In [13]:
from scipy import stats

#1
df['diff'] = (df['bp_pre']-df['bp_post']) 
round(df['diff'].mean(),2)

#2
st, pv = stats.ttest_rel(df['bp_post'], df['bp_pre'], alternative = 'less') # 전후차이일때 사용
print(st, pv)
# st는 통계결과, p-value

-3.0002163948827545 0.0016434474228511028


어떤 특정 약물을 복용한 사람들의 평균 체온이 복용하지 않은 사람들의 평균 체온과 유의미하게 다른지 검정해보려고한다.

검정 통계량, p-value, 검정 결과를 출력

독립표본 T검정: ttest_ind

In [14]:
from scipy import stats

# 데이터 수집
group1 = [36.8, 36.7, 37.1, 36.9, 37.2, 36.8, 36.9, 37.1, 36.7, 37.1]
group2 = [36.5, 36.6, 36.3, 36.6, 36.9, 36.7, 36.7, 36.8, 36.5, 36.7]

In [19]:
sv, pv = stats.ttest_ind(group1, group2)
print(sv, pv)

3.7964208654863336 0.001321891476703691


문제: 다음은 22명의 학생들이 국어시험에서 받은 점수이다. 학생들의 평균이 75보다 크다고 할 수 있는가?¶
귀무가설(H0): 모평균은 mu와 같다. (μ = mu), 학생들의 평균은 75이다
대립가설(H1): 모평균은 mu보다 크다. (μ > mu), 학생들의 평균은 75보다 크다
가정:

모집단은 정규분포를 따른다.
표본의 크기가 충분히 크다.
검정통계량, p-value, 검정결과

단일표본 T검정: ttest_1samp를 출력하시오

In [25]:
from scipy.stats import ttest_1samp

scores = [75, 80, 68, 72, 77, 82, 81, 79, 70, 74, 76, 78, 81, 73, 81, 78, 75, 72, 74, 79, 78, 79]

criteria = 75
cv, pv = ttest_1samp(scores, criteria, alternative = "greater")

print(cv, pv)
# help(ttest_1samp)

1.765879233231226 0.04597614747709146


세 가지 다른 교육 방법(A, B, C)을 사용하여 수험생들의 시험 성적을 개선시키는 효과를 평가하고자 한다. 30명의 학생들을 무작위로 세 그룹으로 배정하여 교육을 실시하였고, 시험을 보고 성적을 측정하였습니다. 다음은 각 그룹의 학생들의 성적 데이터입니다.
귀무가설(H0): 세 그룹(A, B, C) 간의 평균 성적 차이가 없다.
대립가설(H1 또는 Ha): 세 그룹(A, B, C) 간의 평균 성적 차이가 있다.
일원배치법을 수행하여 그룹 간의 평균 성적 차이가 있는지 검정하세요
f값 (소수 둘째자리)
p값 (소수 여섯째자리)
검정결과 출력

In [30]:
import scipy
dir(scipy.stats) # f_oneway
help(scipy.stats.f_oneway)

Help on function f_oneway in module scipy.stats._stats_py:

f_oneway(*samples, axis=0, nan_policy='propagate', keepdims=False)
    Perform one-way ANOVA.
    
    The one-way ANOVA tests the null hypothesis that two or more groups have
    the same population mean.  The test is applied to samples from two or
    more groups, possibly with differing sizes.
    
    Parameters
    ----------
    sample1, sample2, ... : array_like
        The sample measurements for each group.  There must be at least
        two arguments.  If the arrays are multidimensional, then all the
        dimensions of the array must be the same except for `axis`.
    axis : int or None, default: 0
        If an int, the axis of the input along which to compute the statistic.
        The statistic of each axis-slice (e.g. row) of the input will appear in a
        corresponding element of the output.
        If ``None``, the input will be raveled before computing the statistic.
    nan_policy : {'propagate', 'omi

In [34]:
from scipy.stats import f_oneway

# 각 그룹의 데이터
groupA = [85, 92, 78, 88, 83, 90, 76, 84, 92, 87]
groupB = [79, 69, 84, 78, 79, 83, 79, 81, 86, 88]
groupC = [75, 68, 74, 65, 77, 72, 70, 73, 78, 75]

cv, pv = f_oneway(groupA, groupB, groupC)

print(cv, pv)

16.884107946026987 1.752985223798021e-05


12명의 수험생이 빅데이터 분석기사 시험에서 받은 점수이다. Shapiro-Wilk 검정을 사용하여 데이터가 정규 분포를 따르는지 검증하시오
귀무 가설(H0): 데이터는 정규 분포를 따른다.
대립 가설(H1): 데이터는 정규 분포를 따르지 않는다.
Shapiro-Wilk 검정 통계량, p-value, 검증결과를 출력

Shapiro-Wilk 검정는 정규분포를 따르는지 확인하는검정하시오

In [40]:
# dir(scipy.stats)

from scipy.stats import shapiro

data = [75, 83, 81, 92, 68, 77, 78, 80, 85, 95, 79, 89]

cv, pv = shapiro(data)
print(cv, pv)

if pv <= 0.05:
    print("귀무가설 기각")
else:
    print("데이터는 정규분포를 따른다")

0.9768091723993144 0.9676506711851194
데이터는 정규분포를 따른다


iris에서 Sepal Length와 Sepal Width의 상관계수 계산하고 소수 둘째자리까지 출력하시오

In [41]:
import pandas as pd
from sklearn.datasets import load_iris

# iris 데이터셋 로드
iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)

# your code

In [58]:
df.head()

new_df = df[['sepal length (cm)', 'sepal width (cm)']]
print(new_df)

# print(new_df.corr())
print(new_df.corr().loc['sepal length (cm)', 'sepal width (cm)'])

     sepal length (cm)  sepal width (cm)
0                  5.1               3.5
1                  4.9               3.0
2                  4.7               3.2
3                  4.6               3.1
4                  5.0               3.6
..                 ...               ...
145                6.7               3.0
146                6.3               2.5
147                6.5               3.0
148                6.2               3.4
149                5.9               3.0

[150 rows x 2 columns]
-0.11756978413300088


In [45]:
#dir(scipy.stats)

['BootstrapMethod',
 'CensoredData',
 'ConstantInputWarning',
 'Covariance',
 'DegenerateDataWarning',
 'FitError',
 'MonteCarloMethod',
 'NearConstantInputWarning',
 'PermutationMethod',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__path__',
 '__spec__',
 '_ansari_swilk_statistics',
 '_axis_nan_policy',
 '_biasedurn',
 '_binned_statistic',
 '_binomtest',
 '_boost',
 '_bws_test',
 '_censored_data',
 '_common',
 '_constants',
 '_continuous_distns',
 '_covariance',
 '_crosstab',
 '_discrete_distns',
 '_distn_infrastructure',
 '_distr_params',
 '_entropy',
 '_fit',
 '_hypotests',
 '_kde',
 '_ksstats',
 '_levy_stable',
 '_mannwhitneyu',
 '_morestats',
 '_mstats_basic',
 '_mstats_extras',
 '_multicomp',
 '_multivariate',
 '_mvn',
 '_odds_ratio',
 '_page_trend_test',
 '_qmc',
 '_qmc_cy',
 '_qmvnt',
 '_rcont',
 '_relative_risk',
 '_resampling',
 '_rvs_sampling',
 '_sampling',
 '_sensitivity_analysis',
 '_sobol',
 '_stats',

로지스틱 회귀에 대해서...

Pclass, Gender, sibsp, parch를 독립변수로 사용하여 로지스틱 회귀모형을 실시하였을 때, parch변수의 계수값은? 단, Pclass는 범주형 변수이다
(반올림하여 소수 셋째 자리까지 계산)

In [13]:
import pandas as pd
df = pd.read_csv("Titanic.csv")
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Gender   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [28]:
from statsmodels.formula.api import logit
df.columns

formula = "Survived ~ C(Pclass) + Gender + SibSp + Parch"
model = logit(formula, data = df).fit()
model.params

Optimization terminated successfully.
         Current function value: 0.459565
         Iterations 6


Intercept         2.491729
C(Pclass)[T.2]   -0.848152
C(Pclass)[T.3]   -1.866905
Gender           -2.760281
SibSp            -0.232553
Parch            -0.049847
dtype: float64

로지스틱 회귀2


고객 정보를 나타낸 데이터이다. 주어진 데이터에서 500개 중 앞에서부터 350개는 train으로, 150개는 test 데이터로 나눈다. 모델을 학습(적합)할 때는 train 데이터를 사용하고, 예측할 때는 test 데이터를 사용한다. 모델은 로지스틱 회귀를 써서 고객이 특정 제품을 구매할지 여부를 예측하되, 페널티는 부과하지 않는다.

종속변수: purchase (0: 구매 안 함, 1: 구매 함)

문제 1-1. income 변수를 독립변수로 purchase를 종속변수로 사용하여 로지스틱 회귀 모형을 만들고, income 변수가 한 단위 증가할 때 구매할 오즈비 값을 계산하시오. (반올림하여 소수 넷째자리까지 계산)

문제 1-2. 독립변수 income만 사용해 학습한 모델에서 test 데이터의 purchase를 예측하고, accuracy (정확도)를 구하시오. (반올림하여 소수 셋째자리까지 계산)

문제 1-3. 독립변수 income만 사용해 학습한 모델의 로짓 우도를 계산하시오.

문제 1-4. 독립변수 income만 사용해 학습한 모델의 유의확률(p-value)를 구하시오.

In [30]:
import pandas as pd
df = pd.read_csv("Customer_Data.csv")

df.head()

age  income  marital_status  children  gender  purchase
0   62  111980               1         2       1         0
1   65  107314               0         3       1         0
2   18   56209               1         1       1         1
3   21   28010               1         3       0         1
4   21   79618               0         2       0         1

In [50]:
train = df.iloc[:350]
test = df.iloc[350:]

from statsmodels.formula.api import logit

model = logit('purchase ~ income', data = train).fit() # y ~ x
log_odds = model.params['income'] # 로그 오즈 값

import numpy as np
odds = np.exp(log_odds)
print(odds)

pred = model.predict(test)
result = (pred > 0.5).astype(int)

# 정확도 계산
accuracy = (result == test['purchase']).mean()
print(accuracy)

# 유의확률
print(model.pvalues['income'])




Optimization terminated successfully.
         Current function value: 0.692599
         Iterations 3
1.0000019601805765
0.5066666666666667
0.5964910811075123


두 그룹 간 평균 비교

문제: 두 교육 방법의 효과 비교
연구자는 두 가지 다른 교육 방법이 학생들의 성적에 미치는 영향을 비교하고자 합니다. 연구자는 무작위로 선발된 20명의 학생들을 두 그룹으로 나누어 한 그룹에는 교육 방법 A를, 다른 그룹에는 교육 방법 B를 적용합니다. 교육이 끝난 후, 두 그룹의 성적을 비교하기 위해 독립 표본 t-검정과 ANOVA F-검정을 실시하려고 합니다.

다음은 두 그룹의 성적입니다: 다음의 두 가지 검정을 사용하여 두 교육 방법 간의 성적 차이가 통계적으로 유의한지를 검증하세요

독립 표본 t-검정을 실시하여 t 통계량을 구하세요.
독립 표본 t-검정을 실시하여 p-값을 구하세요.
ANOVA F-검정을 실시하여 F 통계량을 구하세요.
ANOVA F-검정을 실시하여 p-값을 구하세요.

In [60]:
import pandas as pd
from scipy.stats import ttest_ind, f_oneway

df = pd.DataFrame({
    'A':[77, 75, 82, 80, 81, 83, 84, 76, 75, 87],
    'B':[80, 74, 77, 79, 71, 74, 78, 69, 70, 72],
})

#1. 독립 표본 t-검정
cv, pv = ttest_ind(df['A'], df['B'])
print(cv, pv)

cv, pv = f_oneway(df['A'], df['B'])
print(cv, pv)



3.1068522301122954 0.006087373605949963
9.652530779753763 0.006087373605949924


In [59]:
(dir(scipy.stats))

['BootstrapMethod',
 'CensoredData',
 'ConstantInputWarning',
 'Covariance',
 'DegenerateDataWarning',
 'FitError',
 'MonteCarloMethod',
 'NearConstantInputWarning',
 'PermutationMethod',
 '__all__',
 '__builtins__',
 '__cached__',
 '__doc__',
 '__file__',
 '__loader__',
 '__name__',
 '__package__',
 '__path__',
 '__spec__',
 '_ansari_swilk_statistics',
 '_axis_nan_policy',
 '_biasedurn',
 '_binned_statistic',
 '_binomtest',
 '_boost',
 '_bws_test',
 '_censored_data',
 '_common',
 '_constants',
 '_continuous_distns',
 '_covariance',
 '_crosstab',
 '_discrete_distns',
 '_distn_infrastructure',
 '_distr_params',
 '_entropy',
 '_fit',
 '_hypotests',
 '_kde',
 '_ksstats',
 '_levy_stable',
 '_mannwhitneyu',
 '_morestats',
 '_mstats_basic',
 '_mstats_extras',
 '_multicomp',
 '_multivariate',
 '_mvn',
 '_odds_ratio',
 '_page_trend_test',
 '_qmc',
 '_qmc_cy',
 '_qmvnt',
 '_rcont',
 '_relative_risk',
 '_resampling',
 '_rvs_sampling',
 '_sampling',
 '_sensitivity_analysis',
 '_sobol',
 '_stats',

카이제곱 적합도 검정
고등학교에서는 졸업생들이 선택하는 대학 전공 분야의 선호도가 시간이 지남에 따라 변하지 않는다고 가정합니다. 학교 측은 최근 졸업생들의 전공 선택이 과거와 같은 패턴을 따르는지 알아보기 위해 적합도 검정을 실시하기로 결정했습니다.

과거 자료에 따르면 졸업생들이 선택하는 전공의 분포는 다음과 같습니다:

인문학: 20% 사회과학: 30% 자연과학: 25% 공학: 15% 기타: 10% 올해 졸업한 학생 200명의 전공 선택 분포는 다음과 같았습니다:

인문학: 30명 사회과학: 60명 자연과학: 50명 공학: 40명 기타: 20명 이 데이터를 바탕으로, 졸업생들의 전공 선택 패턴이 과거와 유사한지를 알아보기 위해 카이제곱 적합도 검정을 실시해야 합니다. 유의 수준은 0.05로 설정합니다.

검정 통계량?
p-value?
유의수준 하 귀무가설 기각 또는 채택?

In [64]:
# 관측된 빈도
observed_frequencies = [30, 60, 50, 40, 20]

# 기대된 빈도 
expected_frequencies = [200 * 0.20, 200 * 0.30, 200 * 0.25, 200 * 0.15, 200 * 0.10]


# 카이제곱
from scipy.stats import chisquare

# dir(scipy.stats)
# help(chisquare)
chisquare(observed_frequencies, expected_frequencies)


Power_divergenceResult(statistic=5.833333333333334, pvalue=0.21194558437271782)

지지도, 신뢰도, 향상도
'빼빼로'와 '딴짓초코'가 함께 팔린 거래의 지지도를 계산하세요.
'빼빼로'가 팔린 거래 중에서 '빼빼로'와 '오징어칩'이 함께 팔린 거래의 신뢰도를 계산하세요.
'빼빼로'와 '양조위빵'의 향상도를 계산하세요.
지지도(A,B): A와 B가 함께 팔린 거래 횟수 / 전체 거래 횟수
신뢰도(A->B): A와 B가 함께 팔린 거래 횟수 / A가 팔린 거래 횟수
향상도(A,B): 신뢰도(A->B) / 지지도(B)

In [65]:
import pandas as pd
# 데이터
df = pd.DataFrame({
    'transaction': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    '빼빼로': [1, 0, 1, 1, 0, 1, 1, 0, 1, 1],
    '딴짓초코': [0, 1, 1, 0, 1, 0, 1, 1, 0, 0],
    '양조위빵': [1, 0, 0, 1, 1, 1, 0, 0, 1, 0],
    '오징어칩': [0, 1, 1, 0, 0, 1, 0, 1, 1, 1],
    '초코파이': [1, 1, 0, 0, 1, 0, 1, 1, 0, 0]
})
df

transaction  빼빼로  딴짓초코  양조위빵  오징어칩  초코파이
0            1    1     0     1     0     1
1            2    0     1     0     1     1
2            3    1     1     0     1     0
3            4    1     0     1     0     0
4            5    0     1     1     0     1
5            6    1     0     1     1     0
6            7    1     1     0     0     1
7            8    0     1     0     1     1
8            9    1     0     1     1     0
9           10    1     0     0     1     0

In [80]:

# 각 문제의 계산을 위한 데이터 준비
total = df.shape[0]
pepero = df['빼빼로'].sum()

# 문제 1: 빼빼로와 딴짓초코가 함께 팔린 거래의 지지도
pepero_and_choco = len(df[(df['빼빼로'] == 1) & (df['딴짓초코'] == 1)])
print(pepero_and_choco / total)

# 문제 2: 빼빼로가 팔린 거래 중 빼빼로와 오징어칩이 함께 팔린 거래의 신뢰도
pepero_and_squid = len(df[(df['빼빼로'] == 1) & (df['오징어칩'] == 1)])
print(pepero_and_squid / pepero)

# 문제 3: 빼빼로와 양조위빵의 향상도
pepero_and_bread = len(df[(df['빼빼로'] == 1) & (df['양조위빵'] == 1)])
bread = df['양조위빵'].sum()
print((pepero_and_bread / pepero) / (bread / total))


0.2
0.5714285714285714
1.1428571428571428


포아송분포
문제: 한 서점에서는 평균적으로 하루에 3명의 고객이 특정 잡지를 구매합니다. 이 데이터는 포아송 분포를 따른다고 가정할 때, 다음 질문에 대한 답을 구하세요.

하루에 정확히 5명의 고객이 잡지를 구매할 확률은 얼마입니까? (%로 값을 정수로 입력하시오)
하루에 적어도 2명의 고객이 잡지를 구매할 확률은 얼마입니까? (%로 값을 정수로 입력하시오)
포아송 분포의 확률 

질량 함수(PMF)image.png

λ는 단위 시간(또는 단위 공간)당 평균 발생 횟수이고, k는 특정 시간(또는 공간) 동안의 이벤트 발생 횟수입니다.

In [91]:
from scipy.stats import poisson

mean = 3

# 하루에 정확히 5명의 고객이 잡지를 구매할 확률
print(poisson.pmf(5, mean))

# 하루에 적어도 2명의 고객이 잡지를 구매할 확률 -> 2 명과 3명
print(1- poisson.cdf(1, mean)) # cdf는 누적 확률을 계산한다.

0.10081881344492458
0.8008517265285442


성별과 시험합격은 독립적인가를 검정하시오!
1 검정 통계량?
2 p-value?
3 귀무가설 기준 (기각/채택)?
4 남자의 합격 기대

chi2는 카이제곱 분포를 나타내는 객체
chi2_contingency는 카이제곱 독립성 검정을 위한 것 빈도?

In [93]:
import pandas as pd

df = pd.DataFrame({'합격': [100, 130], "불합격": [200, 170]}, index = ['남성', '여성'])
df

합격  불합격
남성  100  200
여성  130  170

In [96]:
from scipy.stats import chi2_contingency
chi2_contingency(df)

# help(chi2_contingency)


Chi2ContingencyResult(statistic=5.929494712103407, pvalue=0.01488951060599475, dof=1, expected_freq=array([[115., 185.],
       [115., 185.]]))

베르누이 분포와 이항분포**
[베르누이 분포] 다음 데이터는 100번의 시도에서 각각 성공(1) 또는 실패(0)를 나타냅니다. 이 데이터를 바탕으로 각 시도의 성공 확률을 계산하시오.
[이항분포] 1번 문제에서 계산한 성공 확률을 사용하여, 100번의 시도 중 정확히 60번 성공할 확률을 계산하시오.

In [98]:
df = pd.read_csv("t3_success.csv")
df.head()

Success
0        1
1        0
2        0
3        1
4        1

In [106]:
# 성공확률 
pro = df.sum()/len(df)
print(pro)

from scipy.stats import binom
# 이항분포
binom.pmf(60, 100, pro)

Success    0.62
dtype: float64


array([0.07464986])

점 추정과 구간 추정
데이터셋은 어떤 도시의 일일 평균 온도 입니다.

점추정: 데이터셋을 기반으로 이 도시의 평균 연간 온도를 점추정하세요. (반올림하여 소수 둘째자리까지)
구간추정: 95% 신뢰수준에서 이 도시의 평균 연간 온도에 대한 신뢰구간을 구하세요. (반올림하여 소수 둘째자리까지)

In [107]:
import pandas as pd
temperature_data = pd.read_csv("daily_temperatures.csv")
temperature_data

Daily Average Temperature
0                    28.820262
1                    22.000786
2                    24.893690
3                    31.204466
4                    29.337790
..                         ...
360                  23.492286
361                  20.018854
362                  24.659242
363                  21.699825
364                  19.921589

[365 rows x 1 columns]

In [126]:
#점추정
temperature_data.mean()
sample_std = temperature_data.std()
#구간추정 t.interval

import scipy.stats
# help(scipy.stats.t.interval) # 외우기 어렵다.
scipy.stats.t.interval(0.95, 
                       df = len(temperature_data)-1, 
                       loc = temperature_data.mean(), 
                       scale = sample_std/(len(temperature_data)**0.5))

(array([19.42788709]), array([20.44726799]))

크리스마스 장식 종류와 지역에 따라 판매량에 유의미한 차이가 있는지 이원 분산 분석을 통해 검정하세요
크리스마스 장식 종류(트리, 조명, 장식품)가 판매량에 미치는 영향을 분석하세요. 이때, 장식 종류의 F-value, p-value를 구하시오
지역(북부, 남부, 동부, 서부)이 판매량에 미치는 영향을 분석하세요. 이때, 장식 종류의 F-value, p-value를 구하시오
크리스마스 장식 종류와 지역의 상호작용이 판매량에 미치는 영향을 분석하세요. 이때, 장식 종류의 F-value, p-value를 구시오


In [128]:
import pandas as pd
df = pd.read_csv("christmas_decoration_sales.csv")
df

Decoration_Type Region  Sales
0               트리     북부     64
1               트리     남부     67
2               트리     동부     84
3               트리     서부     87
4               트리     북부     87
5               트리     남부     29
6               트리     동부     41
7               트리     서부     56
8               트리     북부     90
9               트리     남부     32
10              트리     동부     78
11              트리     서부     85
12              조명     북부     59
13              조명     남부     66
14              조명     동부     57
15              조명     서부     45
16              조명     북부     97
17              조명     남부     92
18              조명     동부     29
19              조명     서부     40
20              조명     북부     89
21              조명     남부     99
22              조명     동부     67
23              조명     서부     84
24             장식품     북부     69
25             장식품     남부     49
26             장식품     동부     39
27             장식품     서부     39
28             장식품     북부     34
29             장식품     남부     59
30             장식품     동부     52
31             장식품     서부     85
32             장식품     북부     29
33             장식품     남부     77
34             장식품     동부     52
35             장식품     서부     51

In [2]:
import scipy.stats
help(scipy.stats.t.interval)

Help on method interval in module scipy.stats._distn_infrastructure:

interval(confidence, *args, **kwds) method of scipy.stats._continuous_distns.t_gen instance
    Confidence interval with equal areas around the median.
    
    Parameters
    ----------
    confidence : array_like of float
        Probability that an rv will be drawn from the returned range.
        Each value should be in the range [0, 1].
    arg1, arg2, ... : array_like
        The shape parameter(s) for the distribution (see docstring of the
        instance object for more information).
    loc : array_like, optional
        location parameter, Default is 0.
    scale : array_like, optional
        scale parameter, Default is 1.
    
    Returns
    -------
    a, b : ndarray of float
        end-points of range that contain ``100 * alpha %`` of the rv's
        possible values.
    
    Notes
    -----
    This is implemented as ``ppf([p_tail, 1-p_tail])``, where
    ``ppf`` is the inverse cumulative distribut